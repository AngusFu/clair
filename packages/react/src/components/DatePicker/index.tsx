export * from "./lib/SinglePicker";
export * from "./lib/RangePicker";
