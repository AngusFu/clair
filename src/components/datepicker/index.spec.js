import { mount, createLocalVue } from '@vue/test-utils'
import DatePicker from './index.vue'

// TODO: resolve issue with $clair
const localVue = createLocalVue()
localVue.prototype.$clair = {
  defaultThrottleTime: 150
}

// TODO: more cases
it('should render basic DatePicker', () => {
  let val = ''
  const wrapper = mount(DatePicker, {
    localVue,
    propsData: {
      placeholder: '请输入或者选择日期',
      value: '2017-01-25'
    },
    listeners: {
      change (e) {
        val = e
      }
    }
  })

  expect(wrapper).toMatchSnapshot()
  wrapper.vm.$emit('change', '2018-01-12')
  expect(val).toEqual('2018-01-12')
})
