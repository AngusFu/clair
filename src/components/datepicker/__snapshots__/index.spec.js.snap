// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render basic DatePicker 1`] = `
<div class="c-datepicker">
  <div class="c-datepicker__icon c-datepicker__hovericon"><span class="c-icon">​<svg xmlns="http://www.w3.org/2000/svg" width="1em" height="1em" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-x-circle" style="vertical-align: baseline;"><circle cx="12" cy="12" r="10"></circle><line x1="15" y1="9" x2="9" y2="15"></line><line x1="9" y1="9" x2="15" y2="15"></line></svg></span></div>
  <div clas="" class="c-datepicker__icon"><span class="c-icon">​<svg xmlns="http://www.w3.org/2000/svg" width="1em" height="1em" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-calendar" style="vertical-align: baseline;"><rect x="3" y="4" width="18" height="18" rx="2" ry="2"></rect><line x1="16" y1="2" x2="16" y2="6"></line><line x1="8" y1="2" x2="8" y2="6"></line><line x1="3" y1="10" x2="21" y2="10"></line></svg></span></div>
  <div class="c-input-wrap is-normal"><input placeholder="请输入或者选择日期" type="text" class="c-input">
    <!---->
    <!---->
  </div>
</div>
`;
